### PR TITLE
Add a safe overload for ImageFrame()

### PR DESCRIPTION
**Pull request by sr229**
Resolves GH-16. Will want to migrate tests to this before doing anything so I didn't remove the unsafe overload yet.

### CODE REVIEW

File: Mediapipe.Net/Framework/Format/ImageFrame.cs [hunk 1]
```diff
@@ -49,6 +49,19 @@ unsafe public ImageFrame(ImageFormat format, int width, int height, int widthSte
...
+        public ImageFrame(ImageFormat format, int width, int height, int widthStep, ReadOnlySpan<byte> pixelData)
+        {
+            fixed (byte* pixelDataPtr = pixelData)
+            {
+                UnsafeNativeMethods.mp_ImageFrame__ui_i_i_i_Pui8_PF(
+                    format, width, height, widthStep,
+                    pixelDataPtr,
+                    releasePixelData,
+                    out var ptr).Assert();
+                Ptr = ptr;
```
Comment (Speykious) [inline Mediapipe.Net/Framework/Format/ImageFrame.cs:61]: This is indeed exactly how it's supposed to work, but do you have any solution to remove the code duplication in the 2 constructors? I mean yeah it's just a function followed by an assignment but still.

**Review by Speykious — COMMENTED** [review]
Almost LGTM (I mean in any case I didn't want to remove the old constructor. Maybe sometimes the user does want to be able to pass pointers.), just a tiny small thing

File: Mediapipe.Net/Framework/Format/ImageFrame.cs (hunk 1 above)
Comment (sr229) [reply Mediapipe.Net/Framework/Format/ImageFrame.cs:61]: None that I am aware of, sadly, but we'll cross that bridge when we get to it.

Comment (sr229) [reply Mediapipe.Net/Framework/Format/ImageFrame.cs]: Done via https://github.com/vignetteapp/MediaPipe.NET/pull/20/commits/39dfc6cc5b8a5c722432b8218fe9836893edd494

File: Mediapipe.Net/Framework/Format/ImageFrame.cs [hunk 2]
```diff
@@ -49,6 +49,21 @@ unsafe public ImageFrame(ImageFormat format, int width, int height, int widthSte
             Ptr = ptr;
         }
 
+        public ImageFrame(ImageFormat format, int width, int height, int widthStep, ReadOnlySpan<byte> pixelData) : this(format,
+                                                                                                                         width,
+                                                                                                                         height,
+                                                                                                                         widthStep,
+                                                                                                                         spanToBytePtr(pixelData))
```
Comment (Speykious) [inline Mediapipe.Net/Framework/Format/ImageFrame.cs:56]: ***Excuse me wtf***

File: Mediapipe.Net/Framework/Format/ImageFrame.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public ImageFrame() : base()
 
         public ImageFrame(ImageFormat format, int width, int height) : this(format, width, height, DefaultAlignmentBoundary) { }
 
-        public ImageFrame(ImageFormat format, int width, int height, uint alignmentBoundary) : base()
+        public ImageFrame(ImageFormat format, int width, int height, uint alignmentBoundary)
```
Comment (Speykious) [inline Mediapipe.Net/Framework/Format/ImageFrame.cs:28]: Hmmm. Why remove it?

File: Mediapipe.Net/Framework/Format/ImageFrame.cs [hunk 4]
```diff
@@ -39,7 +39,7 @@ public ImageFrame(ImageFormat format, int width, int height, uint alignmentBound
         //     mediapipe::ImageFormat::Format format,
         //     int width, int height, int width_step, uint8* pixel_data,
         //     Deleter* deleter, mediapipe::ImageFrame** image_frame_out);
-        unsafe public ImageFrame(ImageFormat format, int width, int height, int widthStep, byte* pixelData)
+        unsafe public ImageFrame(ImageFormat format, int width, int height, int widthStep, byte* pixelData) : base()
```
Comment (Speykious) [inline Mediapipe.Net/Framework/Format/ImageFrame.cs:42]: This might be something that we needed to do since the very beginning

**Review by Speykious — CHANGES_REQUESTED** [review]
well

File: Mediapipe.Net/Framework/Format/ImageFrame.cs (hunk 2 above)
Comment (sr229) [reply Mediapipe.Net/Framework/Format/ImageFrame.cs:56]: In my defense, I ran `dotnet format`.

File: Mediapipe.Net/Framework/Format/ImageFrame.cs (hunk 3 above)
Comment (sr229) [reply Mediapipe.Net/Framework/Format/ImageFrame.cs:28]: Wait I removed this by accident, let me re-add that.

**Review by Speykious — APPROVED** [review]
Aight, LGTM

**Pull request merged**